### PR TITLE
Add support for loading private key from file

### PR DESCRIPTION
**Pull request by hs-robertbecker**
## Loading private key from file
We ran into issues with loading the private key inline. These issues were mostly due to our build process, which outputs a malformed YAML file after replacing the secret template in our build files. Nonetheless, having a private key hard inside a file like this, a file that is potentially inside a code repository is not safe.

As a suggestion, I added a `privateKeyFile` config option which just expects a file location. This PR is more of a proof of concept since the UI would also need to be adjusted accordingly.

I am curious about any suggestions!

**Edit**: I am aware that the frontend would need to be updated to reflect the changes in the backend configuration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,22 @@ datasources:
...
+  - name: BigQuery DS
+    type: grafana-bigquery-datasource
+    editable: true
+    enabled: true
+    jsonData:
+      authenticationType: jwt
+      clientEmail: your-client-email
+      defaultProject: your-default-bigquery-project
+      tokenUri: https://oauth2.googleapis.com/token
+      privateKeyFile: '/etc/secrets/bigquery.pem'
```
Comment (dprokop) [inline README.md:83]:
```suggestion
      privateKeyPath: '/etc/secrets/bigquery.pem'
```

**Review by dprokop — COMMENTED** [review]
@hs-robertbecker thanks for the conribution!

I think this is an interesting addition. My suggestions for the UI are as follows:
1. Add an option to fill the JWT manually first. This should show an empty JWT form:


https://user-images.githubusercontent.com/2376619/176451072-d4e00727-feeb-44c1-9310-035b4975bd20.mov


2. In the form allow specifying file path instead of the key itself:

https://user-images.githubusercontent.com/2376619/176450786-1a243c36-9ad4-41fc-b75b-e7003b710b64.mov


@zoltanbedi any thoughts on this?

**Comment by hs-robertbecker** [comment]
I added a new input field like @dprokop suggested. I have to work through the test cases before I will hand in the code for a proper PR
Screen Shot 2022-06-30 at 3 38 45 PM
Screen Shot 2022-06-30 at 3 38 45 PM
Screen Shot 2022-06-30 at 3 39 30 PM
Screen Shot 2022-06-30 at 3 39 30 PM
Screen Shot 2022-06-30 at 3 39 37 PM
Screen Shot 2022-06-30 at 3 39 37 PM

**Comment by hs-robertbecker** [comment]
@dprokop @zoltanbedi 
I am done with the changes, all tests and linting seem to pass!

**Review by zoltanbedi — CHANGES_REQUESTED** [review]
Hi @hs-robertbecker,
Thanks for creating this pr. One more thing, can you please add your change to the CHANGELOG.MD as well. Other than that all looks good and works as expected.

**Comment by hs-robertbecker** [comment]
@zoltanbedi 
✅ CHANGELOG updated
✅  gitignore file was changed to your request

File: yarn.lock [hunk 2]
```diff
@@ -2,35 +2,35 @@
 # yarn lockfile v1
 
 
-"@babel/code-frame@7.10.4":
+"@babel/code-frame@7.10.4", "@babel/code-frame@^7.5.5":
   version "7.10.4"
-  resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.10.4.tgz#168da1a36e90da68ae8d49c0f1b48c7c6249213a"
```
Comment (zoltanbedi) [inline yarn.lock:7]: @hs-robertbecker Can you please revert this change?

File: .gitignore [hunk 3]
```diff
@@ -148,4 +148,4 @@ package-lock.json
 todo.txt
 
 # dist directory
-dist/
\ No newline at end of file
+dist/
```
Comment (zoltanbedi) [inline .gitignore:151]: Revert this as well please.

**Review by zoltanbedi — APPROVED** [review]
Thank you @hs-robertbecker!

**Pull request merged**